# Nomor 3 Pretest : Titanic Dataset

## Load the Required Libraries

In [96]:
import pandas as pd
import numpy as np

## 1. What is the dimension (col, row) of the data frame?

In [97]:
df = pd.read_csv('titanic.csv')

In [4]:
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [5]:
df.tail()

Survived  Pclass                            Name     Sex   Age  \
882         0       2            Rev. Juozas Montvila    male  27.0   
883         1       1     Miss. Margaret Edith Graham  female  19.0   
884         0       3  Miss. Catherine Helen Johnston  female   7.0   
885         1       1            Mr. Karl Howell Behr    male  26.0   
886         0       3              Mr. Patrick Dooley    male  32.0   

     Siblings/Spouses Aboard  Parents/Children Aboard   Fare  
882                        0                        0  13.00  
883                        0                        0  30.00  
884                        1                        2  23.45  
885                        0                        0  30.00  
886                        0                        0   7.75

In [98]:
df.shape

(887, 8)

Jawab :<br>
**887 Rows dan 8 Columns**

## 2. How to know data type of each variable?

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
Survived                   887 non-null int64
Pclass                     887 non-null int64
Name                       887 non-null object
Sex                        887 non-null object
Age                        887 non-null float64
Siblings/Spouses Aboard    887 non-null int64
Parents/Children Aboard    887 non-null int64
Fare                       887 non-null float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.5+ KB


In [7]:
df.dtypes

Survived                     int64
Pclass                       int64
Name                        object
Sex                         object
Age                        float64
Siblings/Spouses Aboard      int64
Parents/Children Aboard      int64
Fare                       float64
dtype: object

Jawab :<br>
**Dapat diketahui terdapat 3 tipe data pada variabel dalam data, yaitu:**<br>
**- Survived, Pclass, Siblings/Spouses Aboard, dan Parents/Children Abroad = integer** <br>
**- Name dan Sex = string** <Br>
**- Age dan Fare = float**

## 3. How many passengers survived (Survived=1) and not-survived (Survived=0)?

In [99]:
df.groupby('Survived').size()

Survived
0    545
1    342
dtype: int64

Jawab:<br>
**Dari hasil diatas dapat dilihat bahwa terdapat :<br>
**- 342 penumpang Survived**<Br>
**- 545 penumpang not-Survived**

## 4. How to drop column ‘Name’ from the data frame?

In [100]:
df.drop(columns=['Name'], inplace=True)

In [101]:
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  
0                        0   7.2500  
1                        0  71.2833  
2                        0   7.9250  
3                        0  53.1000  
4                        0   8.0500

In [102]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 7 columns):
Survived                   887 non-null int64
Pclass                     887 non-null int64
Sex                        887 non-null object
Age                        887 non-null float64
Siblings/Spouses Aboard    887 non-null int64
Parents/Children Aboard    887 non-null int64
Fare                       887 non-null float64
dtypes: float64(2), int64(4), object(1)
memory usage: 48.6+ KB


Jawab:<br>
** Dengan menggunakan syntax 'drop' untuk drop column dan dimensi data sekarang adalah 887 baris dan 7 kolom**

## 5. Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)

In [103]:
df['Family'] = df['Siblings/Spouses Aboard'] + df['Parents/Children Aboard']
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  Family  
0                        0   7.2500       1  
1                        0  71.2833       1  
2                        0   7.9250       0  
3                        0  53.1000       1  
4                        0   8.0500       0

Jawab :<br>
**Query yang dapat digunakan untuk menambah kolom adalah seperti diatas**

## 6. As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for non-missing value). 


In [104]:
#tidak ada missing values
df.isnull().sum()

Survived                   0
Pclass                     0
Sex                        0
Age                        0
Siblings/Spouses Aboard    0
Parents/Children Aboard    0
Fare                       0
Family                     0
dtype: int64

In [120]:
age_miss = []
for i in range(len(df)):
    if np.isnan(df['Age'])[i]:
        age_miss.append('Yes')
    else:
        age_miss.append('No')

In [121]:
df['Age_miss'] = Age_miss
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  Family Age_miss  
0                        0   7.2500       1       No  
1                        0  71.2833       1       No  
2                        0   7.9250       0       No  
3                        0  53.1000       1       No  
4                        0   8.0500       0       No

Jawab:<br>
**Query yang dapat digunakan untuk melihat missing value untuk masing-masing variabel : isnull().sum()**<br>
**Sedangkan, untuk menambah dan insert value pada kolom dapat menggunakan 'loop' dan 'if' statement juga 'append' function **

## 7. Please fill Age missing value with means of existing Age values

In [107]:
mean = df['Age'].mean()
df['Age'].fillna(mean, inplace=True)

Jawab :<br>
**Query yang dapat digunakan adalah seperti diatas**

## 8. What is the maximum passenger Age who survived from the tragedy? 

In [108]:
df[df['Survived']==1]['Age'].max()

80.0

Jawab :<br>
**Usia maksimum dari penumpang yang selamat adalah 80 tahun**

## 9. How many passengers survived from each ‘PClass’? 

In [109]:
df['Pclass'].unique()

array([3, 1, 2], dtype=int64)

In [110]:
pd.DataFrame(df[df['Survived'] == 1].groupby('Pclass').count().Survived)

Survived
Pclass          
1            136
2             87
3            119

Jawab :<br>
**Jumlah penumpang yang selamat untuk masing-masing PClass adalah**<br>
**- 136 orang di kelas 1**<br>
**- 87 orang di kelas 2**<br>
**- 119 orang di kelas 3**

## 10. How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for tttanic1 and 0.3 for titanic2 ? 


In [115]:
titanic1 = df.sample(frac=0.7).reset_index()
titanic1.head()

index  Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0    681         0       2    male  60.0                        1   
1    287         1       3  female  22.0                        0   
2    313         1       3  female  26.0                        0   
3    360         0       3  female  45.0                        0   
4    176         0       1  female  50.0                        0   

   Parents/Children Aboard     Fare  Family Age_miss  
0                        1  39.0000       2       No  
1                        0   7.7500       0       No  
2                        0   7.8542       0       No  
3                        1  14.4542       1       No  
4                        0  28.7125       0       No

In [116]:
titanic2 = df.drop(titanic1.index)
titanic2.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
621         0       3    male  21.0                        0   
622         0       1    male  61.0                        0   
623         0       2    male  57.0                        0   
624         1       1  female  21.0                        0   
625         0       3    male  26.0                        0   

     Parents/Children Aboard     Fare  Family Age_miss  
621                        0  16.1000       0       No  
622                        0  32.3208       0       No  
623                        0  12.3500       0       No  
624                        0  77.9583       0       No  
625                        0   7.8958       0       No

Jawab :<br>
**Query yang dapat digunakan untuk membagi dataframe menjadi 2 bagian dengan proporsi 70:30 adalah seperti diatas**